src/components/App.js

```diff
@@ -1,14 +1,38 @@
-import ButtonPanel from './ButtonPanel';
+import React, { Component } from 'react';
 import Display from './Display';
+import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
 
-calculate(4, 5, '+');
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      lastClicked: null,
+    };
+  }
+
+  handleClick = (buttonName) => {
+    this.setState(({
+      total, next, operation, lastClicked,
+    }) => calculate({
+      total, next, operation, lastClicked,
+    }, buttonName));
+    this.setState({ lastClicked: buttonName });
+  }
```

Comment:
Refactor handleClick like so:
```
 handleClick = (buttonName) => {
    this.setState(({
      total, next, operation,
    }) => calculate({
      total, next, operation,
    }, buttonName));
  }
```
Like the previous reviewer mentioned, you don't need **lastClicked**